scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,390 @@
+        if (ftpUsername == null || ftpPassword == null) {
+            throw new IllegalArgumentException("You first need to add your ftp username and password (FTP_USERNAME, FTP_PASSWORD) to the third party credentials");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException, IllegalArgumentException {
+
+        FTPClient ftpClient = new FTPClient();
+        // connect and login to the server
+        ftpClient.connect(ftpHostname, ftpPort);
+        int reply = ftpClient.getReplyCode();
+        if (!FTPReply.isPositiveCompletion(reply)) {
+            ftpClient.disconnect();
+            throw new IOException("Exception in connecting to FTP Server");
+        }
+        ftpClient.login(ftpUsername, ftpPassword);
+
+        // use local passive mode to pass firewall
+        ftpClient.enterLocalPassiveMode();
+
+        getOut().println("Connected");
+
+        //FTP mode is GET
+        if (ftpMode.equals("GET")) {
+            FtpGet(ftpClient);
+        }
+        //FTP mode is PUT
+        else if (ftpMode.equals("PUT")) {
+            FtpPut(ftpClient);
+        }
+        // log out and disconnect from the server
+        ftpClient.logout();
+        ftpClient.disconnect();
+
+        getOut().println("Disconnected");
+
+        return 1;
+    }
+
+    private void FtpGet(FTPClient ftpClient) throws IOException, IllegalArgumentException {
+        getOut().println("Importing file(s) from " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+        FTPFile[] ftpFile = ftpClient.listFiles(ftpRemoteFilePath);
+        if (ftpFile.length == 0)
+            throw new IllegalArgumentException(ftpRemoteFilePath + " not found. Please, enter a valid path.");
+
+        // If it is a single file:
+        if (ftpFile.length == 1 && ftpRemoteFilePath.contains(ftpFile[0].getName())) {
+            String saveFilePath = Paths.get(ftpLocalFilePath, ftpFile[0].getName()).toString();
+            boolean download = downloadSingleFile(ftpClient, ftpRemoteFilePath, saveFilePath);
+            if (download) {
+                getOut().println("Succeed to download the file " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+            } else {
+                getOut().println("COULD NOT download the file: " + ftpRemoteFilePath);
+            }
+
+            // If the file is a zip, and ftpExtractArchive is set to true
+            if (ftpExtractArchive && ftpRemoteFilePath.endsWith(".zip")) {
+                extractArchive(saveFilePath);
+            }
+        } // If it is a folder, download all its contents recursively
+        else {
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            downloadDirectory(ftpClient, "", "", ftpLocalFilePath);
+        }
+        getOut().println("END Import file(s) from FTP.");
+    }
+
+    private void extractArchive(String saveFilePath) throws IOException {
```

Comment:
Instead of rewriting this method, you can reuse:

org.ow2.proactive.scheduler.common.util.ZipUtils.unzip(ZipFile zipFile, File dest)